packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -589,6 +591,21 @@ export function allocateChildren(vnode: VCustomElement, vm: VM) {
     vm.aChildren = children;
 
     const { renderMode, shadowMode } = vm;
+    if (process.env.NODE_ENV !== 'production') {
+        // If any of the children being allocated is a scoped slot fragment, make sure the receiving
+        // component is a light DOM component. This is mainly to validate light dom parent running
+        // in native shadow mode.
+        if (
+            renderMode !== RenderMode.Light &&
+            children.filter((child) => child && isVScopedSlotFragment(child)).length
```

Comment:
```suggestion
            children.some((child) => child && isVScopedSlotFragment(child))
```

For consistency, I would also suggest using `ArraySome`. (Even though AIUI that's mostly a perf micro-optimization, and this is for debug mode.) Similarly, we could use `!isUndefined(child)`.